README-cmake-details.md

```diff
@@ -0,0 +1,223 @@
+
+If you have selected the "HAL" option, the `build.rs` file will contain a call into the `embuild` library that nakes sure that your component can properly link against - and in general - can "talk" to the [Safe Rust bindings for ESP-IDF](https://github.com/esp-rs/esp-idf-svc) (the "HAL").
+
+#### `components/rust-test/Cargo.toml`:
+
+This is a pretty standard and mostly empty `cargo` [build manifest file](https://doc.rust-lang.org/cargo/reference/manifest.html), which sets up your Rust crate to be built as a Rust static libray. 
+
+If you have selected the "HAL" option when generating the project, the manifest would also list:
+* a dependency to `esp-idf-svc` (i.e. [the safe Rust wrappers for ESP-IDF](https://github.com/esp-rs/esp-idf-svc));
+* a built-time dependency to [`embuild`](https://github.com/esp-rs/embuild) which helps the build integration between ESP-IDF and Rust to run smoothly);
+* the popular Rust [`log`](https://github.com/rust-lang/log) crate.
+
+Post-generation, you can add/remove additional dependencies on Rust crates here as well as modify how Rust compiles your project (as in optimization level and others).
+
+#### `components/rust-test/rust-toolchain.toml`:
+
+This file instructs `cargo` what Rust compiler toolchain to use when building your component. It would list either the `esp` toolchain (a Rust toolchain provided by Espressif that has extra support for the esp32 xtensa MCUs), or the standard Rust `nightly` toolchain. The latter can only be used with Espressif RiscV MCUs.
+
+#### `components/rust-test/src/lib.rs`:
+The actual source code of your Rust library. You can extend here at will.
+
+## Troubleshooting
+
+- If you encounter linker errors, you may need to update your Rust flags. For example, you might need to add the `-Zbuild-std-features=compiler-builtins-weak-intrinsics` flag to `CARGO_BUILD_FLAGS` in your `CMakeLists.txt`.
+
+## Simulation
+
+### Simulation with Wokwi in VS Code
+
+[Wokwi for Visual Studio Code](https://docs.wokwi.com/vscode/getting-started) provides a simulation solution for embedded and IoT system engineers. The extension integrates with your existing development environment, allowing you to simulate your projects directly from your code editor.
+
+- [Install VS Code](https://code.visualstudio.com/download)
+- [Install the Wokwi plugin](https://docs.wokwi.com/vscode/getting-started#installation)
+- Activate Wokwi plugin - command palette, search for `Wokwi: Start Simulator`, select and activate the plugin using web browser
+
+### Add files for Wokwi simulator
+
+Create [`wokwi.toml`](./wokwi.toml) in the root of the project. The file contains references to BIN and ELF previously built by `idf.py`.
+
+```toml
+[wokwi]
+version = 1
+elf = "build/test.elf"
+firmware = "build/test.bin"
+```
+
+Create [`diagram.json`](./diagram.json). The file contains board selected for the simulation.
```

Comment:
Maybe link to the available boards? https://github.com/wokwi/wokwi-boards/tree/main/boards